### ISSUE TITLE
Reconsider to keep the 'classic' mode

### ISSUE THREAD

**Issue by stevefan1999-personal** [nuxt/nuxt.js#4935]
### What problem does this feature solve?

Currently, the exclusion of the so-called Vuex classic mode store creation method will commit in Nuxt 3, and by doing so blocks the usage of [`sascha245/vuex-simple`](https://github.com/sascha245/vuex-simple) that it return a cooked Vuex store.

### What does the proposed changes look like?

So I pledge the Nuxt team to think about letting the classic mode to live on, perhaps in another form like adding a new config in `nuxt.config.js`:
```js
...
export const store = {
  factory: `~/store/createStore.ts`
}
...

```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8596">#c8596</a>)</em></sub></div>

**Comment by Atinux** [comment]
Hi @stevefan1999-personal 

Thank you for this feature request, indeed, we want to deprecate the classic mode to open more possibilities for the `store/` directory.

The idea would be to have a configuration to explain which module do you want to use for the `store/` directory:

`nuxt.config.js`:
```js
export default {
  store: 'vuex-simple-module' // default to `@nuxt/vue-app-store`
}

**Comment by IPRIT** [comment]
Agree with that. My company using classic mode in production and we are built easy to use module system on top of classic mode.

```js
/**
 * @param {Object} modules
 * @param {boolean} namespaced
 * @returns {Object}
 */
export function createVuexModules (modules, namespaced = true) {
  return Object.keys( modules ).reduce((preparedModules, moduleName) => {
    const module = modules[ moduleName ];

    return Object.assign(preparedModules, {
      [moduleName]: {
        namespaced, // should module be namespaced (namespace/action)

        state: resolveDynamicState( module ),
        getters: module.getters || {},
        actions: module.actions || {},
        mutations: module.mutations || {},
        modules: resolveVuexModules( module )
      }
    });
  }, {});
}
```

**Comment by mrjackwills** [comment]
The documentation needs to be improved for the module version, as far as I can tell, you need an index.js file inthe store folder, but this can just be: 

import namedModule from '@/store/nameModule';

and then the namedModule file will be: 

const state () =>{
    someData: undefined,
};

const mutations = {
	doSomething: (state,) => state.someData = 'no longer undefined,
}

export default {state, mutations};

Then nuxt handles creating the store, in such case what's the point of /store/index.js, as all it is is a single line importing the module(s)? Or have I got something terribly wrong

**Comment by shaunc** [comment]
Indeed, there should be some mechanism to consume modules (and maybe stores?...) imported from other places, such as Vuex-simple-based modules. (Eg I have written a library to deal with auth-related state as a Vuex-simple module.)

**Comment by shalldie** [comment]
Is there any way to use `constants` in `modules mode` ?

`modules  mode` has a namespace named by file's name, so i chose `classic mode`, i think tips in editor are important and nice.

It is a bay way to use `string` like `dispatch('someNamespace/someAction')`，`dispatch(types.someAction)` may be fine.

**Comment by sobolevn** [comment]
This is related to https://github.com/wemake-services/wemake-vue-template/issues/1039 in `wemake-vue-template`

I would also like to use `nuxt` with `vuex-simple`, since otherwise I cannot fully rely on my TypeScript typing, which is a bad thing.

Having "classic" `vuex` mode (or something similar like the discussed `factory`) in nuxt is essential here, since otherwise it would not be possible to integrate tools like `vuex-simple`.
